packages/page-accounts/src/Accounts/modals/ProxyOverview.tsx

```diff
@@ -0,0 +1,296 @@
+// Copyright 2017-2020 @polkadot/app-staking authors & contributors
+// This software may be modified and distributed under the terms
+// of the Apache-2.0 license. See the LICENSE file for details.
+import { ApiPromise } from '@polkadot/api';
+import { SubmittableExtrinsic } from '@polkadot/api/types';
+import { AccountId, ProxyType } from '@polkadot/types/interfaces';
+
+import BN from 'bn.js';
+import React, { useState, useEffect } from 'react';
+import { registry } from '@polkadot/react-api';
+import { Button, InputAddress, Modal, TxButton, Dropdown } from '@polkadot/react-components';
+import { useApi } from '@polkadot/react-hooks';
+
+import { useTranslation } from '../../translate';
+import styled from 'styled-components';
+
+interface Props {
+  className?: string;
+  onClose: () => void;
+  previousProxy?: [[AccountId, ProxyType][], BN];
+  proxiedAccount: string;
+}
+
+function createExtrinsic (api: ApiPromise, batchPrevious: any[], batchAdded: any[]): SubmittableExtrinsic<'promise'> | null {
+  console.log('batchPrevious', batchPrevious.toString());
+  console.log('batchAdded', batchAdded.toString());
+
+  if (batchPrevious.length + batchAdded.length === 1) {
+    // eslint-disable-next-line @typescript-eslint/no-unsafe-return
+    return batchPrevious.length
+      ? batchPrevious[0]
+      : batchAdded[0];
+  }
+
+  // eslint-disable-next-line @typescript-eslint/no-unsafe-return
+  // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment
+  return api.tx.utility.batch([...batchPrevious, ...batchAdded]);
+}
+
+function ProxyOverview ({ className, onClose, previousProxy, proxiedAccount }: Props): React.ReactElement<Props> {
+  const { t } = useTranslation();
+  const { api } = useApi();
+  // the stack of extrinsics on the previous proxies
+  const [batchStackPrevious, setBatchStackPrevious] = useState<any[]>([]);
+  // the stack of extrinsics on newly added proxies
+  const [batchStackAdded, setBatchStackAdded] = useState<any[]>([]);
+  // the final extrinsic (either single or a batch)
+  const [extrinsic, setExtrinsic] = useState<SubmittableExtrinsic<'promise'> | null>(null);
+  // list of previous accounts and proxy types
+  const [previousProxyAccountsTypes, _] = previousProxy || [];
+  // actualized list of previous proxy including new changes (deletion if any)
+  const [previousProxyDisplay, setPreviousProxyDisplay] = useState<[AccountId, ProxyType][] | undefined>(previousProxyAccountsTypes);
+  // actualized list of new proxies (additions, if any)
+  const [addedProxies, setAddedProxies] = useState<[AccountId, ProxyType][]>([]);
+
+  useEffect(() => {
+    if (batchStackPrevious.length || batchStackAdded.length) {
+      setExtrinsic(createExtrinsic(api, batchStackPrevious, batchStackAdded));
+    }
+  }, [api, batchStackPrevious, batchStackAdded]);
+
+  const typeOpts = [
```

Comment:
@jacogr can I get this dynamically from the api rather than declaring them myself?